ROADMAP.md

```diff
@@ -1,32 +1,38 @@
-Mission
-======
 
-The mission of Spoon is to provide a high-quality library for analyzing and transforming Java source code.
 
-Design Philosophy
-=================
-
-R1) The Spoon metamodel is as close as possible to the language concepts.
-
-R2) The Spoon model of a program is complete and sound.
-
-R3) The text version of a Spoon model is well-formed and semantically equivalent to the original program.
-
-R4) The analysis and transformation API is intuitive and regular.
-
-R5) Transformation operators are designed to warn as fast as possible about invalid programs. This is done either with static type checking or with dynamic checks when the operators are used.
-
-R6) When feasible, the text version of a Spoon model is close to the original one.
-
-
-Future Work
+Spoon Roadmap
 ==========
 
-The future development of Spoon includes:
-
-* support of Java 7
-* support of Java 8
-* testing framework for transformations
-* keep original indentation for automated maintenance
-
+Short-term, long term and crazy ideas about Spoon
+
+version of 03/09/2015
+
+
+* Model
+    * support for analyzing bound vs unbound type references (`List<T>` vs `List<String>`) 
+* Transformations:
+    * Keep original when printing after transformation
+        * inline comments
+    * indentation and formatting   
+    * sniper mode (only rewrites the changed nodes)
+    * Transactional transformations (rollback if transfo fails)
+    * adds generic transformations that facilitate analysis and transformation
+    * everything in a block
+    * unfinalizer 
+* Processor orchestration 
+    * specify the current strategy
+        * what if multiple processors on the same node
+        * what if new nodes added?
+    * improved version, see 'Source model analysis using the JJTraveler visitor combinator framework'    
```

Comment:
A link to 'Source model analysis using the JJTraveler visitor combinator framework'?